src/python/foglamp/update_statistics_history.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""
+The following piece of code takes the information found in the statistics table, and stores it's delta value 
+(statistics.value - statistics.prev_val) inside the statistics_history table. To complete this, SQLAlchemy will be 
+used to execute SELECT statements against statistics, and INSERT against the statistics_history table.  
+"""
+import sqlalchemy
+import sqlalchemy.dialects
+
+__author__ = "Ori Shadmon"
+__copyright__ = "Copyright (c) 2017 OSI Soft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+# Set variables for connecting to database
+_CONNECTION_STRING = "dbname='foglamp'"
+
+# Deceleration of tables in SQLAlchemy format
+_STATS_TABLE = sqlalchemy.Table('statistics', sqlalchemy.MetaData(),
+                                sqlalchemy.Column('key', sqlalchemy.CHAR(10), primary_key=True),
+                                sqlalchemy.Column('description', sqlalchemy.VARCHAR('255'), default=''),
+                                sqlalchemy.Column('value', sqlalchemy.BIGINT, default=0),
+                                sqlalchemy.Column('previous_value', sqlalchemy.BIGINT, default=0),
+                                sqlalchemy.Column('ts', sqlalchemy.TIMESTAMP(6),
+                                                  default=sqlalchemy.func.current_timestamp()))
+
+"""Description of each column 
+key - Corresponding statistics.key value, so that there is awareness of what the history is of
+history_ts - the newest timestamp in statistics for that key 
+value - delta value between `value` and `prev_val` of statistics
+            The point is to know how many actions happened during the X time
+ts - current timestamp 
+"""
+_STATS_HISTORY_TABLE = sqlalchemy.Table('statistics_history', sqlalchemy.MetaData(),
+                                        sqlalchemy.Column('key', sqlalchemy.CHAR(10)),
+                                        sqlalchemy.Column('history_ts', sqlalchemy.TIMESTAMP(6),
+                                                          default=sqlalchemy.func.current_timestamp()),
+                                        sqlalchemy.Column('value', sqlalchemy.BIGINT, default=0),
+                                        sqlalchemy.Column('ts', sqlalchemy.TIMESTAMP(6),
+                                                          default=sqlalchemy.func.current_timestamp())
+                                        )
+
+
+def __query_execution(stmt="") -> sqlalchemy.engine.result.ResultProxy:
+    """
+    Execute query and return result
+    Args:
+        stmt: Query being executed
+
+    Returns:
+        Result of the query 
+    """
+    
+    engine = sqlalchemy.create_engine('postgres:///foglamp',  pool_size=20, max_overflow=0)
```

Comment:
Update _CONNECTION_STRING to 'postgres:///foglamp' and use _CONNECTION_STRING in call.